src/Provider/Doctrine/Auditing/Transaction/AuditTrait.php

```diff
@@ -163,7 +166,7 @@ private function diff(EntityManagerInterface $entityManager, object $entity, arr
             } elseif (
                 $meta->hasAssociation($fieldName)
                 && $meta->isSingleValuedAssociation($fieldName)
-                && $this->provider->isAuditedField($entity, $fieldName)
+                && ($isAuditedField ?? $this->provider->isAuditedField($entity, $fieldName))
             ) {
```

Comment:
@dmitryuk this does work but I don't like assignments in conditions, I find them less readable.

Please, move this assignment above the `if` and refer to the relevant variable both in the `if` and `elseif` conditions.


```php
           $isAuditedField = $this->provider->isAuditedField($entity, $fieldName);

           if (
                 !isset($meta->embeddedClasses[$fieldName])
                 && $meta->hasField($fieldName)
                 && $isAuditedField
             ) {
                 $mapping = $meta->fieldMappings[$fieldName];
                 $type = Type::getType($mapping['type']);
                $o = $this->value($entityManager, $type, $old);
                $n = $this->value($entityManager, $type, $new);
             } elseif (
                 $meta->hasAssociation($fieldName)
                 && $meta->isSingleValuedAssociation($fieldName)
                 && $isAuditedField
             ) {
  
```

Reply:
I can do it, but the variable $isAuditedField will evaluated in any case even other subconditions returned ’ false’. 
In my solution it doesn't happen

Reply:
@DamienHarper I have reverted these lines, because I think we can cache whole conditions using Doctrine cache and speedup much. Will do it in separate PR.